exercises/bowling/HINTS.md

```diff
@@ -1,12 +1,6 @@
 ## Hints
 
-To complete this exercise, you need to create the data type `Bowling`
-and implement the following functions:
+For a functional language like Haskell it makes sense to calculate the game result in one go from the list of rolls (without an extra `roll` function).
+This result can then either be the total game score or an error for a list of irregular rolls.
 
-- `bowlingStart` returns an empty `Bowling`.
-- `roll` evaluates an input roll, returning the new `Bowling`.
-- `score` calculates the total score at the end if it is regular game.
-
-You will find a dummy data declaration and type signatures already in place,
-but it is up to you to define the functions and create a meaningful data type,
-newtype or type synonym.
+You will find a data declaration for the possible errors as well as the type signature for the `score` function.
```

Comment:
What about not mentioning anything about us diverging from *x-common*?

Maybe just put the instructions and wait until `description.md` is fixed. Something like this:

## Hints

To complete this exercise you need to implement the function `score`,
that takes a sequence of bowling *rolls* and returns the final score or
the appropriate error:

```haskell
score :: [Int] -> Either BowlingError Int

data BowlingError = IncompleteGame
                  | InvalidRoll { rollIndex :: Int, rollValue :: Int }
                  deriving (Eq, Show)
```

You will find these definitions already in place, but it is up to you
to define the function.

Keep in mind that the test suite expects the rolls to be numbered
starting from zero.